internal/operands/data-sources/reconcile.go

```diff
@@ -217,66 +205,254 @@ func reconcileEditRole(request *common.Request) (common.ReconcileResult, error)
+	cronTemplates := sspSpec.CommonTemplates.DataImportCronTemplates
+	cronByDataSource := make(map[client.ObjectKey]cdiv1beta1.DataImportCron, len(cronTemplates))
 	for i := range cronTemplates {
-		cron := &cronTemplates[i]
+		cron := cronTemplates[i].AsDataImportCron()
 		if cron.Namespace == "" {
 			cron.Namespace = internal.GoldenImagesNamespace
 		}
-		cronByDataSource[client.ObjectKey{
-			Name:      cron.Spec.ManagedDataSource,
-			Namespace: cron.Namespace,
-		}] = cron
+		// The architecture annotation should not be in the created DataImportCron.
+		delete(cron.Annotations, DataImportCronArchsAnnotation)
+
+		addToCronMap(cronByDataSource, &cron)
+	}
+
+	return cronByDataSource
+}
+
+func getCronsByDataSourceMultiArch(sourceCollection template_bundle.DataSourceCollection, sspSpec *ssp.SSPSpec) (map[client.ObjectKey]cdiv1beta1.DataImportCron, error) {
+	if !ptr.Deref(sspSpec.EnableMultipleArchitectures, false) {
+		return nil, fmt.Errorf("multi-architecture needs to be enabled")
+	}
+
+	clusterArchs, err := architecture.GetSSPArchs(sspSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	cronByDataSource := map[client.ObjectKey]cdiv1beta1.DataImportCron{}
+	cronTemplates := sspSpec.CommonTemplates.DataImportCronTemplates
+	for i := range cronTemplates {
+		originalCron := cronTemplates[i].AsDataImportCron()
+
+		// Need a copy, because it is modified later.
+		cron := originalCron.DeepCopy()
+		if cron.Namespace == "" {
+			cron.Namespace = internal.GoldenImagesNamespace
+		}
+
+		archsAnnotationValue := cron.Annotations[DataImportCronArchsAnnotation]
+		if archsAnnotationValue == "" {
+			// The ManagedDataSource needs to point to the default DataSource architecture.
+			dsArchs, dsExists := sourceCollection[cron.Spec.ManagedDataSource]
+			if !dsExists {
+				addToCronMap(cronByDataSource, cron)
+				continue
+			}
+
+			defaultArch := getDefaultDataSourceArch(clusterArchs, dsArchs)
+			if defaultArch == "" {
+				// If there is no compatible DataSource architecture, no DataSource is created.
+				addToCronMap(cronByDataSource, cron)
+				continue
+			}
+
+			cron.Spec.ManagedDataSource = cron.Spec.ManagedDataSource + "-" + string(defaultArch)
+			addToCronMap(cronByDataSource, cron)
+			continue
+		}
+
+		// The architecture annotation should not be in the created DataImportCron.
+		delete(cron.Annotations, DataImportCronArchsAnnotation)
+
+		cronArchs := parseArchsAnnotation(archsAnnotationValue)
+		for _, arch := range cronArchs {
+			if !slices.Contains(clusterArchs, arch) {
+				continue
+			}
+
+			cronCopy := cron.DeepCopy()
+			cronCopy.Name = cron.Name + "-" + string(arch)
+			setDataImportCronArchFields(cronCopy, arch)
+			addToCronMap(cronByDataSource, cronCopy)
+		}
+	}
+
+	return cronByDataSource, nil
+}
+
+func addToCronMap(cronMap map[client.ObjectKey]cdiv1beta1.DataImportCron, cron *cdiv1beta1.DataImportCron) {
+	cronMap[client.ObjectKey{
+		Name:      cron.Spec.ManagedDataSource,
+		Namespace: cron.Namespace,
+	}] = *cron
```

Comment:
Make it a map of pointers instead to avoid copies?

Reply:
done